File: sdks/ruby/lib/zerogpu/responses/types/create_response_request.rb
# frozen_string_literal: true

module Zerogpu
  module Responses
    module Types
      class CreateResponseRequest < Internal::Types::Model
        field :model, -> { String }, optional: false, nullable: false
        field :input, -> { Internal::Types::Array[Zerogpu::Types::InputMessage] }, optional: false, nullable: false
        field :text, -> { Zerogpu::Types::TextResponseConfig }, optional: true, nullable: false
      end
    end
  end
end

File: sdks/php/sdk/src/Responses/Requests/CreateResponseRequest.php
<?php

namespace Zerogpu\Responses\Requests;

use Zerogpu\Core\Json\JsonSerializableType;
use Zerogpu\Core\Json\JsonProperty;
use Zerogpu\Types\InputMessage;
use Zerogpu\Core\Types\ArrayType;
use Zerogpu\Types\TextResponseConfig;

class CreateResponseRequest extends JsonSerializableType
{
    /**
     * @var string $model Model identifier from the ZeroGPU dashboard (e.g. summarization or IAB classify).
     */
    #[JsonProperty('model')]
    public string $model;

    /**
     * @var array<InputMessage> $input
     */
    #[JsonProperty('input'), ArrayType([InputMessage::class])]
    public array $input;

    /**
     * @var ?TextResponseConfig $text
     */
    #[JsonProperty('text')]
    public ?TextResponseConfig $text;

    /**
     * @param array{
     *   model: string,
     *   input: array<InputMessage>,
     *   text?: ?TextResponseConfig,
     * } $values
     */
    public function __construct(
        array $values,
    ) {
        $this->model = $values['model'];
        $this->input = $values['input'];
        $this->text = $values['text'] ?? null;
    }
}
